ballista/scheduler/src/state/executor_manager.rs

```diff
@@ -648,6 +648,37 @@ impl ExecutorManager {
             .unwrap_or_else(|| Duration::from_secs(0));
         self.get_alive_executors(last_seen_threshold.as_secs())
     }
+
+    pub(crate) fn get_total_available_task_slots(&self) -> usize {
```

Comment:
This doesn't really give us a realtime view of available task slots since the `available_task_slots` isn't updated in the executor data. Tracking that happens in another data structure. In our case with the redis based `ClusterState` we have a redis map.

Reply:
Hm I hoped this was more "exposable" already. But it seems the ClusterState trait doesn't expose `ExecutorData` yet.